src/cleanlab_tlm/utils/chat.py

```diff
@@ -609,3 +593,248 @@ def _response_to_dict(response: Any) -> dict[str, Any]:
         )
 
     return response.model_dump()
+
+
+def _get_role(message: dict[str, Any]) -> str:
+    if message.get("type", "message") == "message":
+        return cast(str, message.get("role", _USER_ROLE))
+    if message["type"] == _FUNCTION_CALL_TYPE:
+        return _ASSISTANT_ROLE
+    if message["type"] == _FUNCTION_CALL_OUTPUT_TYPE:
+        return _TOOL_ROLE
+    if message["type"] == "file_search_call":
+        return _TOOL_ROLE
+    if message["type"] == "web_search_call":
+        return _TOOL_ROLE
+    return _USER_ROLE
+
+
+def _messages_to_string(messages: list[dict[str, Any]]) -> str:
+    try:
+        from trafilatura import extract
+
+        trafilatura_installed = True
+    except Exception:
+        trafilatura_installed = False
```

Comment:
I think this block should be moved to only right before trafilatura is actually used.

And this block should look more like this at those points:

```
try:
        from trafilatura import extract
    except Exception:
        raise ImportError(
            "The trafilatura package is required to score responses involving web search. Please install it:  `pip install trafilatura`"
        )
```

Reply:
i.e. raise a hard error, rather than raising a warning.